src/NativeScript/SymbolLoader.mm

```diff
@@ -6,75 +6,146 @@
+        CFErrorRef error = nullptr;
+        bool loaded = CFBundleLoadExecutableAndReturnError(this->_bundle.get(), &error);
+        if (error) {
+            dataLogFString([[(NSError*)error description] UTF8String]);
+        }
+
+        return loaded;
+    }
+
+private:
+    WTF::RetainPtr<CFBundleRef> _bundle;
+};
+
+class DlSymbolResolver : public SymbolResolver {
+public:
+    DlSymbolResolver(void* libraryHandle)
+        : _libraryHandle(libraryHandle) {
+    }
+
+    virtual void* loadFunctionSymbol(const char* symbolName) override {
+        return dlsym(this->_libraryHandle, symbolName);
+    }
+
+    virtual void* loadDataSymbol(const char* symbolName) override {
+        return dlsym(this->_libraryHandle, symbolName);
+    }
+
+    virtual bool load() override {
+        return !!this->_libraryHandle;
+    }
+
+private:
+    void* _libraryHandle;
+};
 
 SymbolLoader& SymbolLoader::instance() {
-    static std::once_flag once;
-    static SymbolLoader* loader;
-    std::call_once(once, []() { loader = new SymbolLoader(); });
-    return *loader;
+    static WTF::NeverDestroyed<SymbolLoader> loader;
+    return loader;
 }
 
-CFBundleRef getBundleFromName(std::map<const char*, WTF::RetainPtr<CFBundleRef>>& cache, const char* frameworkName) {
-    auto it = cache.find(frameworkName);
-    if (it != cache.end()) {
-        return it->second.get();
+SymbolResolver* SymbolLoader::resolveModule(const Metadata::ModuleMeta* module) {
+    if (!module) {
+        return nullptr;
     }
 
-    NSString* frameworkPathStr = [NSString stringWithFormat:@"System/Library/Frameworks/%s.framework", frameworkName];
-    NSURL* baseUrl = nil;
+    auto it = this->_cache.find(module);
+    if (it != this->_cache.end()) {
+        return it->second.get();
+    }
 
+    std::unique_ptr<SymbolResolver> resolver;
+    if (module->isFramework()) {
+        NSString* frameworkPathStr = [NSString stringWithFormat:@"%s.framework", module->getName()];
+        NSURL* baseUrl = nil;
+        if (module->isSystem()) {
 #if TARGET_IPHONE_SIMULATOR
-    NSBundle* foundation = [NSBundle bundleForClass:[NSString class]];
-    NSString* foundationPath = [foundation bundlePath];
-    NSString* basePathStr = [foundationPath substringToIndex:[foundationPath rangeOfString:@"/System/Library/Frameworks/Foundation.framework"].location];
-    baseUrl = [NSURL fileURLWithPath:basePathStr
-                         isDirectory:YES];
+            NSBundle* foundation = [NSBundle bundleForClass:[NSString class]];
+            NSString* foundationPath = [foundation bundlePath];
+            NSString* basePathStr = [foundationPath substringToIndex:[foundationPath rangeOfString:@"Foundation.framework"].location];
+            baseUrl = [NSURL fileURLWithPath:basePathStr isDirectory:YES];
+#else
+            baseUrl = [NSURL fileURLWithPath:@"/System/Library/Frameworks" isDirectory:YES];
 #endif
+        } else {
+            baseUrl = [[NSBundle mainBundle] privateFrameworksURL];
```

Comment:
Can you share some frameworks that would need this? I'm just wondering how this is helpful, because I don't remember having anything similar before.